x-pack/plugins/observability/public/pages/alerts/alerts_flyout/alerts_flyout.test.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+import * as DateFormatHook from '../../../hooks/use_date_format';
+
+import { createObservabilityRuleTypeRegistryMock } from '../../../rules/observability_rule_type_registry_mock';
+import { mockHook } from '../../../utils/test_helper';
+import type { TopAlert } from '../';
+import { AlertsFlyout } from './';
+
+describe('AlertsFlyout', () => {
+  mockHook([DateFormatHook, 'useDateFormat'], () => '');
```

Comment:
This might be overkill, but it's a type safe wrapper around jest.spyOn that checks that the mocked property starts with "use" since it's a hook. The type safety is also nice that it checks that 'useDateFormat' is a a property of DateFormatHook import and that it's a function, and then it makes sure the mock is type checked with what the original should do.
Plus I felt that it communicates more clearly about what it aims to do.

Reply:
I'm not 100% sure of this but I think React itself will complain if you try to use something that looks like a hook but doesn't begin with `use`. My feeling is that it's indeed overkill

Reply:
I've removed this.